pkg/clients/openshift.go

```diff
@@ -404,3 +420,57 @@ func (o OpenshiftClient) ListRegistryImages() (images []string, err error) {
 	}
 	return imageList, nil
 }
+
+// GetClusterNetworkPlugin - Get cluster network
+func (o OpenshiftClient) GetClusterNetworkPlugin() (string, error) {
+	net := &networkoapi.ClusterNetwork{}
+	err := o.networkClient.Get().Resource("clusternetworks").Name(networkoapi.ClusterNetworkDefault).Do().Into(net)
+	if err != nil {
+		return "", err
+	}
+	return net.PluginName, nil
+}
+
+// GetNetNamespace - Get Net Namespace.
+func (o OpenshiftClient) GetNetNamespace(nsName string) (*networkoapi.NetNamespace, error) {
+	netNamespace := &networkoapi.NetNamespace{}
+	err := o.networkClient.Get().Resource("netnamespaces").Name(nsName).Do().Into(netNamespace)
+	if err != nil {
+		return nil, err
+	}
+	return netNamespace, nil
+}
+
+// JoinNamespacesNetworks - Will take the net namespace to be added to a network,
+// and the namespace ID of the network that is being added to.
+func (o OpenshiftClient) JoinNamespacesNetworks(netns *networkoapi.NetNamespace,
+	targetNS string,
+) (*networkoapi.NetNamespace, error) {
+	if netns.Annotations == nil {
+		netns.Annotations = make(map[string]string)
+	}
+	netns.Annotations[ChangePodNetworkAnnotation] = fmt.Sprintf("%s:%s", "join", targetNS)
+	result := &networkoapi.NetNamespace{}
+	err := o.networkClient.Put().Resource("netnamespaces").Name(netns.Name).Body(netns).Do().Into(result)
+	if err != nil {
+		return nil, err
+	}
+	return result, nil
```

Comment:
Why return ```result``` if it's not used by the caller?

Reply:
no good reason. eventually, you may need it and you can ignore it when you call the method. 

do you have strong feelings on this one?

Reply:
It doesn't matter to me if you think it will be used eventually I'm fine with it.

I would double check though that ```result``` isn't going to be the same as the```netns``` param.  I would have to do some testing to verify, but the I don't think ```NetName``` and ```NetId``` will change.